### PR TITLE
add `enum`

### PR DESCRIPTION
**Pull request by mflatt**
As an example,

```
HorizAlignment:
  left
  center
  right
```

defines `HorizAlignment` as an annotaton that is satisfied by `#'left`, `#'center`, and `#'right`, and it also defines `HorizAlignment.left`, `HorizAlignment.center`, and `HorizAlignment.right` as expressions bound to those symbols and as binding forms that match those symbols.

Sometimes, enumerated symbols are combined with other datatypes. Although a union of an enumerated datatype with other annotations could cover that case, `enum` provides direct support through `~is_a` clauses that add other annotations directly to the enumeration. Here's an example:

```
enum LineWidth:
  ~is_a Real
  inherit
```

The `pict` and `slideshow` libraries now use `enum` for various enumerations instead of a hand-rolled annotation.

### CODE REVIEW

File: rhombus/scribblings/ref-enum.scrbl [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  defn.macro 'enum $id_name:
+                $enum_clause
+                ...'
+  grammar enum_clause:
+    $id ...
+    ~is_a: $annot; ...
+    ~is_a $annot
+){
+
+ Defines @rhombus(id_name) as a @defterm{predicate annotation} and as a
```
Comment (usaoc) [inline rhombus/scribblings/ref-enum.scrbl:18]: Should be `@term`.

File: rhombus/scribblings/ref-stxobj-meta.scrbl [hunk 2]
```diff
@@ -14,8 +14,8 @@
                         at_stx :: Syntax)
     :: None
   fun syntax_meta.error(message :: ReadableString,
-                        at_stx :: Syntax,
-                        in_stx :: Syntax || List.of(Syntax))
+                        in_stx :: Syntax,
+                        at_stx :: Syntax || List.of(Syntax))
```
Comment (usaoc) [inline rhombus/scribblings/ref-stxobj-meta.scrbl:18]: This shouldn’t be included, I think.  (I see that there’s some problem in this doc, but I’ll fix it in a separate commit.)

**Comment by usaoc** [comment]
I pushed a commit that adds some simple test cases for the new `enum` spacer.

File: rhombus/scribblings/ref-enum.scrbl (hunk 1 above)
Comment (mflatt) [reply rhombus/scribblings/ref-enum.scrbl]: Fixed.

File: rhombus/scribblings/ref-stxobj-meta.scrbl (hunk 2 above)
Comment (mflatt) [reply rhombus/scribblings/ref-stxobj-meta.scrbl]: Removed.

**Comment by mflatt** [comment]
> I pushed a commit that adds some simple test cases for the new `enum` spacer.

Thanks!

**Pull request merged**